### DISCUSSION TITLE
What I would have liked to know before moving to zwave-js from a classic HA-system

### DISCUSSION THREAD

**Discussion by lurendrejer** [zwave-js/node-red-contrib-zwave-js#144] [Ideas]
I don't know if it is worth mentioning on the 'what is it?'-part of the github page.

I had a very big concern when moving to/testing out the zwave-js plugin: Will it mess up my network like domoticz did when I tried that out for fun. Domoticz used som sort of 'template' where endpoint x-something is always associated to domoticz to help the user out - I have had similar headaches when trying out openhab.
"We don't change anything unless you tell us to" - kind of a deal.

I had a small 'oh shoot'-moment when it came to the security/network keys - but my network doesn't run S1 or S2, so I could just leave them out. My network suffered no damages while being moved from homeseer and I could work in node-red/zwave-JS for an hour, shut it down and restart homeseer to have my home working again when the kids walked in the door.

**Comment by marcus-j-davies** [comment]
So,

Endpoints are a real big pain in the behind and then some - every manufacturer has implemented them  in like 1 of 5000 ways.
The specifications don't help detail what should be correct.

The developer behind the javascript driver that this module uses has been working tirelessly to get the right balance.

You can read about it here:
https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v8

I think during the device interview the **Lifeline** associations are scanned, sorted and fixed, to allow correct lifeline associations to the controller (i.e the purpose of the Lifeline association group)

I don't know if associations between specific devices are touched, during the interview.

Let's test that.......

I created a device-2-device association + removed the lifeline (I felt brave)
reinterviewed the device, and now the lifelines are corrected, but my device-2-device association, has remained.

I.e this is good!

Don't get me wrong, there are some exotic requirements with some devices, but these are dealt with via compatiblity flags in the device configuration files.
